File: python/vedant_bhatnagar/DevsnestProblems_dailyUpdated/nextgreaterelement.py
def solve(n, nums):
    # CODE HERE
    nums += nums
    st = [nums[-1]]
    nge = [-1]*n

    for i in range((n*2)-2,-1,-1):
        index = i%n
        while st:
            if (nums[i] >= st[-1]):
                st.pop()
            else:
                break
        if not st:
            nge[index] = -1
        else:
            nge[index] = st[-1]
        st.append(nums[i])
    return nge

File: DN DSA THA/Sandhya tha/Next Greater Element.cpp
 
vector<int> solve(int n, vector<int> nums){

    vector<int> res(n);
    stack<int> st;
    for(int i=2*n-1;i>=0;i--)
    {
        while(!st.empty() && nums[i%n]>=st.top())
                st.pop();
        
            if(st.empty()) res[i%n] = -1;
            else res[i%n] = st.top();
        
        st.push(nums[i%n]);
    }
    return res;
}
